src/charm.py

```diff
@@ -235,7 +317,15 @@ def _on_stop(self, event):
                 # Log but ignore error since we're stopping anyway.
                 logger.exception("Failed to clear runners")
 
-    def _reconcile_runners(self, runner_manager):
+    def _reconcile_runners(self, runner_manager: RunnerManager) -> Dict[str, "JsonObject"]:
```

Comment:
Out of curiosity, what is `JsonObject`?

Reply:
It is a type defined by ops framework to represent JSON-line object in Python.
For example, the juju action result could JSON, so the typing for the method to set the action result involves `JsonObject`.

In ops framework:
```
class ActionEvent(EventBase):
...
    def set_results(self, results: '_SerializedData'):
...
```
with:
```_SerializedData = Dict[str, 'JsonObject']```